server/src/controllers/seasonTicketTypePriceDefaultController.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable camelcase */
+import {Router, Request, Response} from 'express';
+import {checkJwt, checkScopes} from '../auth';
+import {Prisma} from '@prisma/client';
+import {extendPrismaClient} from './PrismaClient/GetExtendedPrismaClient';
+import {InvalidInputError} from './eventInstanceController.service';
+
+const prisma = extendPrismaClient();
+export const seasonTicketTypePriceDefaultController = Router();
+
+// All further routes require appropriate authentication
+seasonTicketTypePriceDefaultController.use(checkJwt);
+seasonTicketTypePriceDefaultController.use(checkScopes);
+
+interface SeasonTicketTypePriceDefaultRequestItem {
+    tickettypeid_fk: number;
+    price: number;
+    concessionprice: number;
+}
+
+/**
+ * @swagger
+ * /2/season-ticket-type-price-default/{id}:
+ *   get:
+ *     summary: get season ticket type prices for a given season
+ *     tags:
+ *     - Season Ticket Type Price Default API
+ *     parameters:
+ *     - $ref: '#/components/parameters/id'
+ *     security:
+ *       - bearerAuth: []
+ *     responses:
+ *       200:
+ *         description: array of updated season ticket type prices
+ *         content:
+ *           application/json:
+ *             schema:
+ *              $ref: '#/components/schemas/SeasonTicketTypePriceDefault'
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *              $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+seasonTicketTypePriceDefaultController.get('/:id', async (req: Request, res: Response) => {
+  try {
+    const {id} = req.params;
+    const toSend = await prisma.seasontickettypepricedefault.findMany({
+      where: {
+        seasonid_fk: +id,
+      },
+    });
+    return res.json(toSend);
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      return res.status(400).send({error: error.message});
+    }
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      return res.status(400).send({error: error.message});
+    }
+    return res.status(500).send({error: 'Internal Server Error'});
+  }
+});
+
+
+/**
+ * @swagger
+ * /2/season-ticket-type-price-default/{id}:
+ *   put:
+ *     summary: update season ticket type prices for a given season
+ *     tags:
+ *     - Season Ticket Type Price Default API
+ *     parameters:
+ *     - $ref: '#/components/parameters/id'
+ *     security:
```

Comment:
How do you specify whether you are updating, adding or deleting?

Reply:
The season ticket type prices are updated to match the array sent with the request body.

- If the ticket type exists in the table for the season and is not in the array it is deleted.
- If the ticket type is in the array but doesn't exist in the table for the given season it is created.
- If the ticket type both exists in the table for the given season and array it is updated.

Reply:
Ok, that makes sense, do you think you could add the three cases to the swagger docs above. It looks like its possible to describe multiple request bodies.
![image](https://github.com/WonderTix/WonderTix/assets/92124260/bc45ae60-96f6-4b6b-ae92-3929d50027d4)

Reply:
The image is from https://swagger.io/docs/specification/describing-request-body/

Reply:
Added examples